code/HabitTracker/app/src/main/java/com/example/habittracker/activities/LoginActivity.java

```diff
@@ -207,6 +212,20 @@ public void registerPressed(View view) throws InvalidKeySpecException, NoSuchAlg
         //DatabaseManager.get().getUsersColName()
 
         String username = usernameField.getText().toString();
+
+        if (username.length() == 0) {
+            Toast.makeText(LoginActivity.this, "Username cannot be empty!", Toast.LENGTH_SHORT).show();
+            return;
+        }
+        if (passwordField.getText().toString().length() < 6) {
+            Toast.makeText(LoginActivity.this, "Password must be at least 6 characters long!", Toast.LENGTH_SHORT).show();
+            return;
+        }
+        if (username.equals(passwordField.getText().toString())) {
+            Toast.makeText(LoginActivity.this, "Username cannot be identical to the password!", Toast.LENGTH_SHORT).show();
```

Comment:
What's wrong with username being identical to the password?

Reply:
It is a very simple security check to make sure the password isn't super weak. We could go much further and make sure the username isn't a substring in the password, that the password isn't among a list of popular passwords, that the password contains a variety of different characters, etc.

Reply:
Okay, cool. I'll approve the PR!